module/zfs/zle.c

```diff
@@ -74,10 +74,14 @@ zle_decompress(void *s_start, void *d_start, size_t s_len, size_t d_len, int n)
 	while (src < s_end && dst < d_end) {
 		int len = 1 + *src++;
 		if (len <= n) {
+			if (src + len > s_end || dst + len > d_end)
```

Comment:
Check my math on this, but shouldn't this be `>=` and the same before.

Reply:
Assuming `src + len == s_end`, after running `*dst++ = *src++` `len` times, `src` becomes `s_end`. Which is fine, because `*src++` dereference before addition.